api/coreservice.go

```diff
@@ -1372,62 +1370,53 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
 	return res
 }
 
-func (core *coreService) LogsInBlockByHash(filter *logfilter.LogFilter, blockHash hash.Hash256) ([]*iotextypes.Log, error) {
+func (core *coreService) LogsInBlockByHash(filter *logfilter.LogFilter, blockHash hash.Hash256) ([]*action.Log, error) {
 	blkHeight, err := core.dao.GetBlockHeight(blockHash)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, "invalid block hash")
 	}
 	return core.logsInBlock(filter, blkHeight)
 }
 
-// LogsInBlock filter logs in the block x
-func (core *coreService) LogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
-	return core.logsInBlock(filter, blockNumber)
-}
-
-func (core *coreService) logsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
-
+func (core *coreService) logsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*action.Log, error) {
 	logBloomFilter, err := core.bfIndexer.BlockFilterByHeight(blockNumber)
 	if err != nil {
 		return nil, err
 	}
 
 	if !filter.ExistInBloomFilterv2(logBloomFilter) {
-		return []*iotextypes.Log{}, nil
+		return []*action.Log{}, nil
 	}
 
 	receipts, err := core.dao.GetReceipts(blockNumber)
 	if err != nil {
 		return nil, err
 	}
 
-	h, err := core.dao.GetBlockHash(blockNumber)
-	if err != nil {
-		return nil, err
-	}
-
-	return filter.MatchLogs(receipts, h), nil
+	return filter.MatchLogs(receipts), nil
```

Comment:
better fill in `log.BlkHash` here, b/c `logsInBlock()` is used by couple of other places